web/packages/teleport/src/config.ts

```diff
@@ -260,7 +260,12 @@ const cfg = {
   },
 
   getSsoUrl(providerUrl, providerName, redirect) {
-    return cfg.baseUrl + generatePath(providerUrl, { redirect, providerName });
+    const params = new URLSearchParams();
+
+    params.set('connector_id', providerName);
+    params.set('redirect_url', redirect);
+
+    return cfg.baseUrl + providerUrl + `?${params.toString()}`;
```

Comment:
According to the spec `URLSearchParams.set` will append the value to the list which is exactly what we want. However there is nothing here guaranteeing that someone doesn't come by and invert the order of those sets, or maybe even add another one to the end. So I think we should have some tests that explicitly test this order.